recipes/mysql-connector-cpp/all/conanfile.py

```diff
@@ -0,0 +1,258 @@
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "17")
+
+        # Compiler minimum version
+        compiler = self.settings.compiler
+        compiler_name = str(compiler)
+        minimum_version = self._minimum_compilers_version.get(compiler_name, False)
+        if minimum_version and Version(compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"Requires compiler {compiler_name} minimum version: {minimum_version} with C++17 support."
+            )
+
+    def requirements(self):
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("zstd/[>=1.5 <1.6]")
+        self.requires("lz4/[>=1.9.2 <=1.10.0]")
+        self.requires("protobuf/3.21.12")
+        self.requires("rapidjson/1.1.0")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.24 <4]")
+        self.tool_requires("protobuf/<host_version>")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def _package_folder_dep(self, dep, scope="host"):
+        try:
+            if scope == "build":
+                return self.dependencies.build[dep].package_folder.replace("\\", "/")
+        except:
+            self.output.info(f"Failed to get a build scoped package path for : {dep}")
+        return self.dependencies[dep].package_folder.replace("\\", "/")
+
+    def _include_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.includedirs[0].replace("\\", "/")
+
+    def _lib_folder_dep(self, dep):
+        return self.dependencies[dep].cpp_info.libdirs[0].replace("\\", "/")
+
+    def generate(self):
+
+        tc = CMakeToolchain(self)
+
+        # OpenSSL
+        tc.cache_variables["WITH_SSL"] = self._package_folder_dep("openssl")
+        tc.cache_variables["OPENSSL_ROOT_DIR"] = self._package_folder_dep("openssl")
+        # LZ4 patches
+        tc.cache_variables["WITH_LZ4"] = self._package_folder_dep("lz4")
+        # ZLIB patches
+        tc.cache_variables["WITH_ZLIB"] = self._package_folder_dep("zlib")
+        # ZSTD patches
+        tc.cache_variables["WITH_ZSTD"] = self._package_folder_dep("zstd")
+        # Build patches
+        tc.cache_variables["BUILD_STATIC"] = not self.options.shared
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        # Disable Boost, only legacy JDBC connector needs it
+        tc.cache_variables["BOOST_DIR"] = "FALSE"
```

Comment:
why not support legacy JDBC? in this PR
https://github.com/conan-io/conan-center-index/pull/17594
i added support to it. 

i think there is still useful to support the old JDBC connector for services that are using mysql server 6x